articles/prototyping-machine-learning-models-with-streamlit/index.md

```diff
@@ -0,0 +1,185 @@
+The installation is very simple, just run this on your command prompt or terminal
+
+```python
+Pip install streamlit
+Streamlit hello
+```
+To import it, use the simple
+
+```python
+import pandas as pd
+import streamlit as st
+import plotly.express as px
+```
+And to run your streamlit app simply use
+
+```python
+Streamlit run app.py
+```
+#### Building Sentiment Analyzer
+Now for this introductory tutorial, we will be building a sentiment analyzer model and prototype it with streamlit. The first step will be getting a Dataset, and the dataset we will be using can be found HERE, next let’s create a directory and create a virtual environment.
+
+```python
+$ mkdir textanalyzer
+$ cd textanalyzer
+```
+Now to create the Virtual environment use the code below
+
+```python
+$ python3.8 -m venv env
+$ source env/bin/activate
+```
+Now copy the dataset as it is in the folder and paste it into this textanalyzer directory, now create a file and save it as sentiment_analyzer, this will be where we will be writing our code to build the streamlit app. The next thing will be to import all the packages we will be using for this tutorial.
+
+```python
+import streamlit as st
+from sklearn.feature_extraction.text import CountVectorizer
+from sklearn.naive_bayes import BernoulliNB
+from sklearn.metrics import confusion_matrix
+import numpy as np
+import itertools
+import matplotlib.pyplot as plt
+```
+After that let’s add a header for our web app and a subheader and to do that we simply use the code below
+
+```python
+st.title("Sentiment Analyzer Based On Text Analysis ")
+st.subheader("Paras Patidar - MLAIT")
+st.write('\n\n')
+```
+The next step will be to write a function that will get all the data from our directory and bring them up,
+
+```python
+@st.cache
+def get_all_data():
+    root = "Datasets/"
+    with open(root + "imdb_labelled.txt", "r") as text_file:
+        data = text_file.read().split('\n')
+         
+    with open(root + "amazon_cells_labelled.txt", "r") as text_file:
+        data += text_file.read().split('\n')
+
+    with open(root + "yelp_labelled.txt", "r") as text_file:
+        data += text_file.read().split('\n')
+
+    return data
+```
+If you notice in the function above the first line is *@st.cache*, this mark tells streamlit that whenever the function is called that there are some certain requirements it should check for and they should be met, now create a variable called all_data which should call the *get_all_data* function we just wrote.
+
+```python
+all_data = get_all_data()
+```
+Next, we create a checkbox using the *st.checkbox* function that will display Show dataset and this checkbox should have the value from the *all_data* variable we created earlier.
+
+```python
+if st.checkbox('Show Dataset'):
+    st.write(all_data)
+```
+Now we write another function that will carry out data preprocessing and to achieve that we use the code below.
+
+```python
+@st.cache
+def preprocessing_data(data):
```

Comment:
Initially, when you talk about the dataset, using pandas.DataFrame, can you load the dataset into a data frame and perform df.head() operation? It will give a glance at what the dataset is.

Reply:
the dataset is just a mere 3 text file in a folder and might not need any peak since we're not building a model the conventional way.